sdk/core/core-amqp/src/errors.ts

```diff
@@ -676,7 +677,7 @@ export function translate(err: unknown): MessagingError | Error {
     return error;
   }
 
-  if (err instanceof Error && err.name === "MessagingError") {
+  if (isError(err) && err.name === "MessagingError") {
```

Comment:
Could we just use instanceof here? Is it important to handle all errors named MessagingError and not just bona fide instances of the class?

```suggestion
  if (err instanceof MessagingError) {
```

Reply:
core-amqp is doing things around errors so I was trying to not change the runtime behavior of existing code.  But it looks that tests are passing so I updated it.

Reply:
Some change failed service-bus tests. Checking...